app/controllers/fhir/r4/api_controller.rb

```diff
@@ -300,12 +300,12 @@ def create
 
         Rails.logger.info "Created Patient with ID: #{resource.id}"
 
-        # Send enrollment notification only to responders
-        resource.send_enrollment_notification if resource.self_reporter_or_proxy?
-
         # Create a history for the enrollment
         History.enrollment(patient: resource, created_by: @current_actor_label, comment: 'Monitoree enrolled via API.')
       end
+
+      # Send enrollment notification only to responders
+      resource.send_enrollment_notification if resource.self_reporter_or_proxy?
```

Comment:
might need to add a check to see if resource is actually a thing (since the transaction may roll back)

Reply:
Yeah good call. I assumed rollback would always come with an error and so we'd never reach here, but it turns out there is one error type, `ActiveRecord::Rollback`, that is not re-raised by a transaction block. TIL.

I added raise statement to get the behavior we want. I just amended the commit since it's so small: https://github.com/SaraAlert/SaraAlert/pull/804/commits/c1eddcadfaf17ae985b5389b6fec3f5ec27c0829

Reply:
I like it!